src/unix/pipe.c

```diff
@@ -272,6 +279,13 @@ int uv_pipe_connect2(uv_connect_t* req,
   if (flags & UV_PIPE_NO_TRUNCATE)
     if (namelen > sizeof(saddr.sun_path))
       return UV_EINVAL;
+  /*
+   * _PC_NAME_MAX returns the maximum length of a filename in the directory
+   * path that the process is allowed to create
+   * https://linux.die.net/man/3/pathconf
+   */
+  if (namelen > (size_t) pathconf(".", _PC_NAME_MAX))
+    return UV_EINVAL;
```

Comment:
This needs to be under the
```
if (flags & UV_PIPE_NO_TRUNCATE)
```
condition a few lines up so that truncation can still occur if `UV_PIPE_NO_TRUNCATE` is not set.